### PR TITLE
DAOS-14693 test: Skip MD on SSD stages by default

### PR DESCRIPTION
**Pull request by phender**
To ease CI queues disable running MD on SSD functional Hardware stages by default in PRs. To enable running theses stages the submitter will need to select the appropriate build parameter or pass a matching 'Skip-func-hw-*: false' commit pragma.

Also adding large and medium verbs provider MD on SSD stages to match the daily-testing branch.

Skip-unit-tests: true
Skip-fault-injection-test: true
Allow-unstable-test: true
Test-tag: always_passes

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -155,11 +155,11 @@ pipeline {
...
-        string(name: 'TestNvme',
-               defaultValue: '',
-               description: 'The launch.py --nvme argument to use for the Functional test ' +
-                            'stages of this run (i.e. auto, auto_md_on_ssd, auto:-3DNAND, ' +
-                            '0000:81:00.0, etc.)')
+        // string(name: 'TestNvme',
+        //        defaultValue: '',
+        //        description: 'The launch.py --nvme argument to use for the Functional test ' +
+        //                     'stages of this run (i.e. auto, auto_md_on_ssd, auto:-3DNAND, ' +
+        //                     '0000:81:00.0, etc.).  Does not apply to MD on SSD stages.')
```
Comment (daltonbohning) [inline Jenkinsfile:162]: Is this meant to be removed in the master branch?

File: Jenkinsfile [hunk 2]
```diff
@@ -262,12 +256,18 @@ pipeline {
         booleanParam(name: 'CI_medium-verbs-provider_TEST',
                      defaultValue: true,
                      description: 'Run the Functional Hardware Medium Verbs Provider test stage')
+        booleanParam(name: 'CI_medium-verbs-provider-md-on-ssd_TEST',
+                     defaultValue: true,
```
Comment (daltonbohning) [inline Jenkinsfile:260]: Doesn't this mean it's on by default?

File: Jenkinsfile [hunk 3]
```diff
@@ -282,19 +282,16 @@ pipeline {
...
                defaultValue: 'ci_nvme5',
-               description: 'Label to use for 5 node Functional Hardware Medium Verbs Provider stage')
+               description: 'Label to use for 5 node Functional Hardware Medium Verbs Provider (MD on SSD) stages')
         string(name: 'FUNCTIONAL_HARDWARE_MEDIUM_UCX_PROVIDER_LABEL',
                defaultValue: 'ci_ofed5',
                description: 'Label to use for 5 node Functional Hardware Medium UCX Provider stage')
         string(name: 'FUNCTIONAL_HARDWARE_LARGE_LABEL',
                defaultValue: 'ci_nvme9',
-               description: 'Label to use for 9 node Functional Hardware Large tests')
+               description: 'Label to use for 9 node Functional Hardware Large (MD on SSD) stages')
```
Comment (daltonbohning) [inline Jenkinsfile:294]: Are these for MD on SSD or the regular stages? Or both?

File: Jenkinsfile (hunk 1 above)
Comment (phender) [reply Jenkinsfile:162]: Yes, with there being separate MD on SSD stages that have this value fixed to auto_md_on_ssd it is not really needed for the other stages and could end up running duplicate stages with the same --nvme setting if set to auto_md_on_ssd.

File: Jenkinsfile (hunk 3 above)
Comment (phender) [reply Jenkinsfile:294]: Both

File: Jenkinsfile (hunk 1 above)
Comment (brianjmurrell) [reply Jenkinsfile:162]: Should the commented out code just be removed?

File: Jenkinsfile [hunk 4]
```diff
@@ -194,11 +194,9 @@ pipeline {
         string(name: 'CI_EL8_TARGET',
                defaultValue: '',
                description: 'Image to used for EL 8 CI tests.  I.e. el8, el8.3, etc.')
-        /* pipeline{} is too big for this
```
Comment (brianjmurrell) [inline Jenkinsfile:197]: Thank-you, thank-you, thank-you!  I meant to try this again at some point.

**Review by brianjmurrell — APPROVED** [review]

**Comment by phender** [comment]
> 'Skip-func-hw-*: false'

https://daosio.atlassian.net/browse/DAOS-14726

**Pull request merged**